### PR TITLE
- Separate out the tail of the EventLoop into separate method to ease…

### PR DESCRIPTION
**Pull request by baldersheim**
… readability.

- Also separate out timeout handling of old connections to separate method and unify on vespalib::time.

@havardpe PR

### CODE REVIEW

File: fnet/src/vespa/fnet/transport_thread.cpp [hunk 1]
```diff
@@ -19,6 +19,7 @@ using vespalib::ServerSocket;
 using vespalib::SocketHandle;
 using vespalib::SocketSpec;
 using OptimizeFor = vespalib::Executor::OptimizeFor;
```
Comment (havardpe) [inline fnet/src/vespa/fnet/transport_thread.cpp:21]: We no longer need OptimizeFor

Comment (baldersheim) [reply fnet/src/vespa/fnet/transport_thread.cpp]: Fixed

File: fnet/src/vespa/fnet/transport_thread.h [hunk 2]
```diff
@@ -156,6 +154,9 @@ class FNET_TransportThread : public FastOS_Runnable
      **/
     bool InitEventLoop();
 
+    void endEventLoop();
```
Comment (havardpe) [inline fnet/src/vespa/fnet/transport_thread.h:157]: should not be public

File: fnet/src/vespa/fnet/transport_thread.h [hunk 3]
```diff
@@ -156,6 +154,9 @@ class FNET_TransportThread : public FastOS_Runnable
...
+    void checkTimedoutComponents(vespalib::duration timeout);
```
Comment (havardpe) [inline fnet/src/vespa/fnet/transport_thread.h:158]: should not be public

File: fnet/src/vespa/fnet/transport_thread.h (hunk 2 above)
Comment (baldersheim) [reply fnet/src/vespa/fnet/transport_thread.h:157]: It is in the private: section so I think this is good.

File: fnet/src/vespa/fnet/transport_thread.h (hunk 3 above)
Comment (baldersheim) [reply fnet/src/vespa/fnet/transport_thread.h:158]: It is in the private: section so I think this is good.

Comment (havardpe) [reply fnet/src/vespa/fnet/transport_thread.h:158]: this should not be together with the event loop external API, then

File: fnet/src/vespa/fnet/transport_thread.h (hunk 2 above)
Comment (havardpe) [reply fnet/src/vespa/fnet/transport_thread.h:157]: this should not be together with the event loop external API, then

Comment (havardpe) [reply fnet/src/vespa/fnet/transport_thread.h:157]: my bad, I forgot we removed the single-threaded option a long time ago

File: fnet/src/vespa/fnet/transport_thread.h (hunk 3 above)
Comment (havardpe) [reply fnet/src/vespa/fnet/transport_thread.h:158]: my bad, I forgot we removed the single-threaded option a long time ago

**Pull request merged**